utils/src/main/java/org/web3j/utils/Numeric.java

```diff
@@ -227,15 +228,18 @@ private static String toHexStringZeroPadded(BigInteger value, int size, boolean
     }
 
     public static String toHexString(byte[] input, int offset, int length, boolean withPrefix) {
-        StringBuilder stringBuilder = new StringBuilder();
-        if (withPrefix) {
-            stringBuilder.append("0x");
-        }
-        for (int i = offset; i < offset + length; i++) {
-            stringBuilder.append(String.format("%02x", input[i] & 0xFF));
-        }
+        final String output = new String(toHexCharArray(input, offset, length, withPrefix));
+        return withPrefix ? new StringBuilder(HEX_PREFIX).append(output).toString() : output;
+    }
 
-        return stringBuilder.toString();
+    public static char[] toHexCharArray(byte[] input, int offset, int length, boolean withPrefix) {
```

Comment:
why is `withPrefix` still here?

Reply:
It's to preserve the functionality of toHexString and to avoid an interface change.

Reply:
but `toHexCharArray` is only defined in this class and could be private

Reply:
Yes, you're right. `toHexCharArray` could be made private to preserve the interface as it is.

Reply:
If you make this update I'm happy to merge.

Reply:
I reviewed and tested the suggestion for performance improvement from @esaulpaugh but found none. I have nothing else to add to this pull, it's good to merge if you approve

Reply:
what about the removal of `withPrefix`?